akka-http-core/src/main/scala/akka/http/impl/util/StreamUtils.scala

```diff
@@ -223,39 +225,36 @@ private[http] object StreamUtils {
    * completion or error signal. It doesn't necessarily mean, though, that all of a streaming pipeline
    * is finished, only that the part that contains this flow has finished work.
    */
-  def identityFinishReporter[T]: Flow[T, T, Future[Unit]] = {
-    // copy from Sink.foreach
-    def newForeachStage(): (PushStage[T, T], Future[Unit]) = {
-      val promise = Promise[Unit]()
+  def identityFinishReporter[T]: Flow[T, T, Future[Done]] = {
```

Comment:
Couldn't this just be replaced by using `Flow.watchTermination` in the call sites instead of this custom stage?